src/aggregation/mod.rs

```diff
@@ -154,7 +156,108 @@ use columnar::{ColumnType, MonotonicallyMappableToU64};
 pub(crate) use date::format_date;
 pub use error::AggregationError;
 use itertools::Itertools;
-use serde::{Deserialize, Serialize};
+use serde::de::{self, Visitor};
+use serde::{Deserialize, Deserializer, Serialize};
+
+/// deserialize Option<f64> from string or float
+pub(crate) fn deserialize_option_f64<'de, D>(deserializer: D) -> Result<Option<f64>, D::Error>
+where D: Deserializer<'de> {
+    struct StringOrFloatVisitor;
+
+    impl<'de> Visitor<'de> for StringOrFloatVisitor {
+        type Value = Option<f64>;
+
+        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            formatter.write_str("a string or a float")
+        }
+
+        fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+        where E: de::Error {
+            value.parse::<f64>().map(Some).map_err(|_err| {
```

Comment:
we should refuse "nan"

Reply:
elastic search allows NaN and Infinity

Reply:
What is the use case for parsing NaN I don't understand?

Reply:
It could be kind of a safeguard, if you would e.g. expect in your avg aggregation that every doc has at least one value, you could set `missing: NaN`, so the avg aggregation would be set to `NaN` if there are missing values.
There may be other queries like this.

Reply:
Not worth the trouble. Let's forbid NaN.

Reply:
I think this still accepts NaN and inf.